module2/homework/grayscaleImages/compression.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <array>
+#include <vector>
+
+constexpr uint32_t height = 32; //160;
+constexpr uint32_t width = 32; //240;
+
+void printMap (std::array<std::array<uint8_t, width>, height> bitmap);
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale (std::array<std::array<uint8_t, width>, height> bitmap);
+
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(std::vector<std::pair<uint8_t, uint8_t>> compressed);
```

Comment:
You can simplify it like this :)
```suggestion
using Image = std::array<std::array<uint8_t, width>, height>;

using Compressed = std::vector<std::pair<uint8_t, uint8_t>>;

Compressed compressGrayscale(const Image& compress);

Image decompressGrayscale(const Compressed& compressedMap);

void printMap(const Image& mapToPrint);
```

Reply:
I recommend using aliases like Vicx95 is suggesting.